packages/frontend/core/src/hooks/use-stable.ts

```diff
@@ -0,0 +1,9 @@
+import { useRef } from 'react';
+
+export const useReadOnce = <T>(fn: () => T): T => {
```

Comment:
I think this is the same with `const [v] = useState(fn); return v;`;

Reply:
excuse me? `const a = useReadOnce(fn) === a = fn()`

Reply:
the same, `const a = useMemo(() => v.a, [v])`  equals `a = v.a`

Reply:
[![confused](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/WjlTemxb6oA4PgZFmj08/9403235f-2f1e-48f1-a6fc-2a3a04a41ea4/confused.gif)](https://app.graphite.dev//settings/meme-library?org=toeverything)

Reply:
I think the use case of this hook is to make sure fn is only invoked once only even its ref is changed

Reply:
whatever

Reply:
never used... deleted